src/lender/tranche/tranche.sol

```diff
@@ -61,23 +62,28 @@ contract Tranche is DSNote, Auth {
         return token.totalSupply();
     }
 
-    // -- Lender Side --
+    /// supplies currency to the tranche and mints new tranche token in exchange
+    /// the exchange rate is defined by the ward contract
     function supply(address usr, uint currencyAmount, uint tokenAmount) external note auth {
         require(currency.transferFrom(usr, self, currencyAmount), "currency-transfer-failed");
         token.mint(usr, tokenAmount);
     }
 
+    /// redeems currency from a tranche in exchange for tranche tokens
+    /// burns the tranche token
+    /// the exchange rate is defined by the ward contract
```

Comment:
' exchange rate is defined by the ward contract' i don't think this is entirely accurate, it is determined by the tranche asset value and how much has been invested

Reply:
yes, but from the perspective of tranche contract, it is defined by another contract.
how it is done, exactly might be confusing here for the reader

Reply:
that was my thinking

Reply:
I agree with your point manuel. I would limit it to the functionality that is relevant to the code here.